### ISSUE TITLE
Add Methods to Sun and MoonPosition

### ISSUE THREAD

**Issue by discovererer** [MenoData/Time4J#740]
hi
Could you add some methodes like:
```
//moment of moon arrive at the location
public static Moment getMoment(double rightAscension, double declination){ }
```
and more  astronomy methods

```
public static boolean isAt(Zodiac.SCORPIA){ } //isMoonAt(Zodiac.SCORPIA) or isSunAt(Zodiac.SCORPIA)
public static Moment getEntryMoment(Zodiac.SCORPIA) { }
public static Moment getExitMoment(Zodiac.SCORPIA) { }
```
https://en.wikipedia.org/wiki/Zodiac
https://en.wikipedia.org/wiki/88_modern_constellations

**Comment by MenoData** [comment]
Sounds good and is a very nice enhancement. Thanks for feedback. However, at the moment, I am still busy with the Chinese calendar and cannot do several things at the same time but will consider your request not so far in the future.

By the way, the Chinese calendar has something similar to zodiac called solar terms so this should be really doable to add the zodiac methods wanted.

About `public static Moment getMoment(double rightAscension, double declination){ }`, I think the result can only be an `Optional<Moment>` which can be empty if the given position expressed in celestial coordinates is not possible at all. On the other side, the sun (or moon) can be several times on the same position so an additional parameter of type `Moment` might be required. Example:

`public static Optional<Moment> atPositionAfter(Moment after, double rightAscension, double declination)`

Feel free to suggest better method or parameter names. I also don't know just now how to implement it in detail so I need some time to think...

**Comment by MenoData** [comment]
More links about zodiac:

// containing data about zodiacs and ecliptic longitudes
https://de.wikipedia.org/wiki/Zodiak
http://star.arm.ac.uk/orrery/zodiac.html
http://earthsky.org/astronomy-essentials/dates-of-suns-entry-into-each-constellation-of-the-zodiac

// containing data about constellation boundaries in celestial coordinates (as polygons)
https://www.iau.org/public/themes/constellations/

A conversion between ecliptic coordinates and celestial coords (RA+decl) is necessary taking into account the boundaries defined by IAU.

**Comment by MenoData** [comment]
When thinking again about your first proposal 

`public static Moment getMoment(double rightAscension, double declination) {...}`

I see heavy implementation difficulties if the exact full position is passed. It does not seem to be very useful, either, because the user can be forced to test many positions before he/she can find the associated moment (just because the sun or moon might never reach that position).

How about a variation where the user only passes one component of the position (for example the azimuth)? Then it is probably possible to determine the moment even if the sun or moon is below the horizon. See for comparison also that issue:

https://github.com/mourner/suncalc/issues/25

Would be glad to hear your opinion.

**Comment by MenoData** [comment]
I have now released v4.37 which contains all zodiac-related calculations, even with precession or for horoscopic signs. About the other wish:

```
//moment of moon arrive at the location
public static Moment getMoment(double rightAscension, double declination){ }

```
I think that a) it is not really useful (because it requires the user to know that the sun or moon will really reach that exact position) and b) it is rather out of scope (because - in contrast to events like sunrise/set, moonrise/set - I don't see any relationship to calendrical issues).

**Comment by discovererer** [comment]
Thanks a lot for this enhancement
